### PR TITLE
Fix: allow tooltip disabled icon button

### PR DESCRIPTION
**Pull request by klaascuvelier**
Closes carbon-design-system/carbon-components-angular#3049

Fix regression from v4: this will not hide the tooltip for a disabled icon-button.

#### Changelog

**Changed**

* fix: allow tooltip for disabled icon-button

### CODE REVIEW

File: src/button/icon-button.component.ts [hunk 1]
```diff
@@ -27,7 +27,6 @@ import { ButtonSize, ButtonType } from "./button.types";
 	<cds-tooltip
 		class="cds--icon-tooltip"
 		[description]="description"
-		[disabled]="disabled"
```
Comment (Akshat55) [inline src/button/icon-button.component.ts:30]: I think we should continue to keep this, this was intentional. So we should create a an input 
```ts
@Input() showTooltipWhenDisabled = false
```


```html
[disabled]="showTooltipWhenDisabled ? true : disabled"
```

File: src/button/icon-button.stories.ts [hunk 2]
```diff
@@ -134,3 +134,33 @@ WithAutoAlign.args = {
...
+			<svg class="cds--btn__icon" cdsIcon="copy" size="16"></svg>
+		</cds-icon-button>
+	`
+});
+export const DisabledWithTooltip = DisabledWithTooltipTemplate.bind({});
+DisabledWithTooltip.args = {
+	autoAlign: true,
+	description: "Icon Description",
+	align: "top"
+};
```
Comment (Akshat55) [inline src/button/icon-button.stories.ts:166]: Probably don't need a separate story to showcase this

**Review by Akshat55 — CHANGES_REQUESTED** [review]

File: src/button/icon-button.stories.ts (hunk 2 above)
Comment (klaascuvelier) [reply src/button/icon-button.stories.ts:166]: Was for me to make testing easier, but will remove :)

File: src/button/icon-button.component.ts (hunk 1 above)
Comment (klaascuvelier) [reply src/button/icon-button.component.ts:30]: The logic in your example was off btw, `showTooltipWhenDisabled` should be `false` in the ternary :D

**Comment by klaascuvelier** [comment]
Both remarks have been addressed @Akshat55, not sure if you want me to resolve them or you rather do it yourself after checking it out.

**Pull request merged**